File: backend/src/main/java/com/tutict/finalassignmentbackend/model/news/TrafficNewsArticle.java
package com.tutict.finalassignmentbackend.model.news;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TrafficNewsArticle {

    private String title;
    private String description;
    private String content;
    private String url;
    private String image;
    private String publishedAt;
    private String sourceName;
    private String sourceUrl;
}

File: flutter_app/lib/features/model/traffic_news_article.dart
class TrafficNewsArticleModel {
  final String? title;
  final String? description;
  final String? content;
  final String? url;
  final String? image;
  final DateTime? publishedAt;
  final String? sourceName;
  final String? sourceUrl;

  const TrafficNewsArticleModel({
    this.title,
    this.description,
    this.content,
    this.url,
    this.image,
    this.publishedAt,
    this.sourceName,
    this.sourceUrl,
  });

  factory TrafficNewsArticleModel.fromJson(Map<String, dynamic> json) {
    return TrafficNewsArticleModel(
      title: json['title'],
      description: json['description'],
      content: json['content'],
      url: json['url'],
      image: json['image'],
      publishedAt: json['publishedAt'] != null
          ? DateTime.tryParse(json['publishedAt'])
          : null,
      sourceName: json['sourceName'],
      sourceUrl: json['sourceUrl'],
    );
  }
}
